File: PrimerCorte/MatrizEjercicios/Reto3/Reto3.cpp
#include <iostream>

using namespace std;

int main(){

    //Declarar filas y columnas
    int columnas = 3, filas = 2;

    //Crear matriz con el tamaño anterior
    int matrizMapa[filas][columnas] = {{1, 2, 3},
                                       {4, 5, 6}};

    // Crear matriz invertida para guardar el resultado
    int matriz[columnas][filas];
    

    // Recorrer cada una de las columnas de la matriz original
    for(int i = 0; i < columnas; i++){
        
        // Recorrer cada una de las filas de la matriz original
        for(int j = 0; j < filas; j++){
            
            //Guardar el valor inverso en una variable
            int valor = matrizMapa[filas - 1 - j][i];
            
            // darle el valor a la posicion inversa de la matriz
            matriz[i][j] = valor;
        }
    }

//For para imprimir la matriz volteada
for(int f = 0; f < columnas; f++){
    for(int c = 0; c < filas; c++){
        cout << "[" << matriz[f][c] << "]" << " ";
    }
    cout << endl;
}
return 0;

}

File: PrimerCorte/MatrizEjercicios/Reto3/Reto3.py
"""RETO 3  ·  Rotar el plano 90 grados   
Contexto: Un mapa de zonas de reciclaje está guardado como matriz, pero se necesita imprimirlo girado para que coincida con la orientación real del barrio.
Se pide: Escribir una función que reciba una matriz de f filas por c columnas y devuelva una nueva matriz rotada 90 grados en sentido horario. Verificar que [1,2,3],[4,5,6] 
produce [4,1],[5,2],[6,3].
Pista: La matriz resultante tiene c filas y f columnas: las dimensiones se intercambian. El elemento de la posición (i,j) termina en la posición (j, f-1-i)."""

#Crear matriz original
matrizMapa=[[1,2,3],
            [4,5,6]]


#Declarar filas y columnas
columnas= len(matrizMapa[0])
filas= len(matrizMapa)

#Crear matriz nueva
matriz = []

# Llenar la matriz
for i in range(columnas):
    fila_actual = []
    for j in range(filas):

        #Usar la formula y guardarla en un valor y luego en la fila
        valor = matrizMapa[filas-1-j][i]
        fila_actual.append(valor)

    #Agregar la fila doble a la matriz
    matriz.append(fila_actual)
    
print(matriz)